### PR TITLE
[Integration tests] Taking offer on auctions

### PR DESCRIPTION
**Pull request by tomaszslabon**
[Probably not needed - do not merge]

This pull request adds three integration tests:
- `when auction fully filled before auction length elapsed`. Scenario: Auction fully filled in one shot before the auction length elapsed. Assert the correct amount has been seized from the pool and what the bidder received. 
-  `when auction fully filled after auction length elapsed`.Scenario: Auction fully filled after the auction length elapsed. Assert the whole pool was seized and transferred to the bidder.
- `when auction fully filled with multiple take offers`.Scenario: Auction fully filled but using multiple partial take offers. Assert the correct amount has been seized from the pool and what bidders received.

### CODE REVIEW

File: test/integration/liquidation.test.js [hunk 1]
```diff
@@ -105,6 +138,188 @@ describe("Integration -- liquidation", () => {
...
+      // velocityPoolDepletingRate = 1.0 (initial value)
+      // amountOnOffer = (timeSinceAuctionStart - startTimeOffset)
+      //   * velocityPoolDepletingRate / auctionLength =
+      //   (4320 - 0) * 1.0 / 86400 = 0.05
+      // portionToSeize = amountOnOffer * amountToTransfer / amountOutstanding =
+      //   0.05 * (4 * 10^18) / (10 * 10^18 ) = 0.02
+      // amountInAssetPool = 100 * 10^18
+      // amountToSeize = amountInAssetPool * portionToSeize =
+      //   (100 * 10^18) * 0.02 = 2 * 10^18
+      // Bidder should receive (2 * 10^18) of collateral tokens
```
Comment (tomaszslabon) [inline test/integration/liquidation.test.js:215]: Posting all these calculations here feels like an overkill, but I wanted to show how we arrived with the values of collateral tokens seized from asset pool that we expect to get in test cases.

Comment (pdyraga) [reply test/integration/liquidation.test.js:215]: > Alternatively, we could just check that after the auction was fully filled, the total number of collateral tokens the bidders received is equal to the initial amount of collateral tokens in the asset pool.

Plus, we could check that what they received is what the auction offered, especially if bidding through `AuctionBidder.takeOfferWithMin`. 

I do agree those calculations are overkill, I would leave them for unit tests.

**Review by lukasz-zimnoch — APPROVED** [review]
LGTM! @tomaszslabon please address the remaining comment and I think we can merge.

**Comment by lukasz-zimnoch** [comment]
Please note the changes introduced by https://github.com/keep-network/coverage-pools/pull/88 before merging. It changed the way signer bonds are handled.

**Comment by tomaszslabon** [comment]
@pdyraga 
As we modified the [system test scenarios](https://github.com/keep-network/coverage-pools/issues/83), I think this PR is no longer needed and should be closed without merging.

Testcase `when auction fully filled before auction length elapsed` will be covered by scenario 1. from the table (the system test already exists but will need to be updated).

Testcase `when auction fully filled with multiple take offers` will be covered by scenario 4 (we decided to simplify the flow - take offers after auction elapsed with two bidders, the scenario was too complicated for a system test). The person who takes scenario 4. will decide whether to write an integration test or a system test.

Testcase `when auction fully filled after auction length elapsed` will also be covered by scenario 4.

What do you think?

**Comment by pdyraga** [comment]
This PR is going to be replaced by #114, correct @tomaszslabon ?

**Comment by tomaszslabon** [comment]
> This PR is going to be replaced by #114, correct @tomaszslabon ?

@pdyraga 
Yes, it will be replaced by #114 (I'm finishing it in a moment - encountered some problem with calculations), and by another PR that adds a system test number 4 from [this list](https://github.com/keep-network/coverage-pools/issues/83#issuecomment-860604097).

**Comment by lukasz-zimnoch** [comment]
Closing in favor of https://github.com/keep-network/coverage-pools/pull/114

**Pull request closed without merging**